scaaml/capture/aes/crypto_alg.py

```diff
@@ -0,0 +1,205 @@
+"""AES specific SCryptoAlgorithm."""
+from typing import Dict
+from typing import Literal
+import numpy as np
+
+from scaaml.capture.aes.acqktp import AcqKeyTextPatternScaaml as ktp_scaaml
+from scaaml.io import resume_kti
+
+
+class SCryptoAlgorithm:
+    """attack points and maybe some basic information about it (e.g. key length, etc.)"""
+    def __init__(self,
+                 firmware_sha256: str,
+                 crypto_implementation,
+                 purpose: Literal['train', 'test', 'holdout'],
+                 implementation: str = "MBEDTLS",
+                 algorithm: str = "simpleserial-aes",
+                 keys: int = 3072,
+                 plaintexts: int = 256,
+                 repetitions: int = 1,
+                 examples_per_shard: int = 64,
+                 full_kt_filename: str = 'key_text_pairs.txt',
+                 full_progress_filename: str = 'progress_pairs.txt') -> None:
+        """Generates a set of key-text pairs and saves those. Does not overwrite
+        existing files.
+
+        Args:
+          firmware_sha256: SHA256 hash of the binary used on the chip.
+          crypto_implementation: The class that provides attack points info and
+            attack points values (for instance scaaml.aes_forward.AESSBOX).
+          purpose: Type of the dataset. Used in scaaml.io.Dataset.
+          implementation: Name of the implementation that was used.
+          algorithm: Algorithm name.
+          keys: Number of different keys that are used.
+          plaintexts: Number of different plaintexts used with each key.
+          repetitions: Number of captures with a concrete (key, plaintext) pair.
+          examples_per_shard: Size of a single part (for ML training purposes).
+          kt_filename: Filename to save key, text pairs (using resume_kti).
+          progress_filename: Filename to save progress (using resume_kti).
+          full_kt_filename: The file to save key-text pairs into. This should
+            be in the same directory as the whole dataset.
+          full_progress_filename: The file to save progress into. This should
+            be in the same directory as the whole dataset.
+        """
+        # TODO(karelkral): Automatically check scaaml.io.utils.sha256sum(firmware_file).
+        # When changing the following assert update also
+        # SCryptoAlgorithm._dataset
+        assert purpose in ['train', 'test', 'holdout']
```

Comment:
Considering how often this is used, it might me worth having it declared in a `constants.py` file.
Not blocking for this PR though. Just something to keep in mind for further code cleaning.

Reply:
Noted in issue #38 (same below)